homeassistant/components/influxdb.py

```diff
@@ -165,9 +171,15 @@ def influx_event_listener(event):
                 except (ValueError, TypeError):
                     new_key = "{}_str".format(key)
                     json_body[0]['fields'][new_key] = str(value)
-                    if non_digit_tail.match(json_body[0]['fields'][new_key]):
-                        json_body[0]['fields'][key] = float(
-                            non_decimal.sub('', value))
+                    if non_digit_tail.match(
+                            json_body[0]['fields'][new_key]) and not (
+                                isinstance(value, datetime.datetime)):
+                        try:
+                            json_body[0]['fields'][key] = float(
+                                non_decimal.sub('', value))
+                        except (ValueError, TypeError):
+                            json_body[0]['fields'][key] = float(
```

Comment:
@philhawthorne, sorry if I was not understood, I was referring to this except clause, to simply `pass` if a `ValueError` is raised.